### PR TITLE
Reroot surface attach fix

### PR DESCRIPTION
**Pull request by whale2**
This should fix issue #15
https://github.com/Tidal-Stream/ProceduralParts/issues/15

Works for me in 1.6.1

### CODE REVIEW

File: Source/ProceduralPart.cs [hunk 1]
```diff
@@ -1452,6 +1452,17 @@ public void PartParentChanged(Part newParent)
 				childToParent = part.FindAttachNodeByPart(newParent);
 			}
 
+            bool srfAttached = false;
+			if (childToParent == null && newParent != null)
```
Comment (Tidal-Stream) [inline Source/ProceduralPart.cs]: This code looks a bit weird, are you using tabs here?

File: Source/ProceduralPart.cs [hunk 2]
```diff
@@ -1479,7 +1490,13 @@ public void PartParentChanged(Part newParent)
             // ReSharper disable once InconsistentNaming
             Func<Vector3> Offset;
             if (nodeOffsets.TryGetValue(childToParent.id, out Offset))
+            {
                 position -= Offset();
+            }
+            else if(srfAttached)
+            {
+                position -= (childToParent.attachedPart.transform.position - childToParent.owner.transform.position);
```
Comment (Tidal-Stream) [inline Source/ProceduralPart.cs:1497]: There is a regression reported on the forum, it could be from this line.

**Review by Tidal-Stream — COMMENTED** [review]
Thanks for the contribution!

File: Source/ProceduralPart.cs (hunk 1 above)
Comment (whale2) [reply Source/ProceduralPart.cs]: True that, looks fine in IDE, didn't notice in git diff output.

File: Source/ProceduralPart.cs (hunk 2 above)
Comment (whale2) [reply Source/ProceduralPart.cs:1497]: Hopefully fixed, waiting for other players to test.

File: Source/ProceduralPart.cs [hunk 3]
```diff
@@ -1452,16 +1452,28 @@ public void PartParentChanged(Part newParent)
 				childToParent = part.FindAttachNodeByPart(newParent);
 			}
 
+            bool srfAttached = false;
+            if (retry && childToParent == null && newParent != null)
```
Comment (MikeOnTea) [inline Source/ProceduralPart.cs]: var parentIsSurfaceAttached = false;
            if (childToParent == null && newParent?.srfAttachNode?.attachedPart == part)
            {
                childToParent = newParent.srfAttachNode;
                parentIsSurfaceAttached = true;
            }
Should work and not require the ugly retry?

Comment (whale2) [reply Source/ProceduralPart.cs]: Indeed that's much better. Just checked, works pretty good.

**Comment by Tidal-Stream** [comment]
Much time passed without reports about issues so I think I can merge it in mainline. Can you revert the .dll and .csproj changes so I can merge it?

**Comment by whale2** [comment]
@Tidal-Stream 
That wasn't really intended. Fixed now.

**Comment by Tidal-Stream** [comment]
Thank you! I am merging and will release a beta release when I can get access to KSP and a compiler.

**Comment by Tidal-Stream** [comment]
[Procedural Parts 1.3.19 beta](https://github.com/Tidal-Stream/ProceduralParts/releases/tag/v1.3.19b)

**Pull request merged**